sycl/source/device_selector.cpp

```diff
@@ -34,11 +36,23 @@ static bool isDeviceOfPreferredSyclBe(const device &Device) {
          backend::level_zero;
 }
 
+static bool isDeviceOfPreferredNumber(detail::device_filter_list *FilterList,
+                                      int index) {
+  for (const detail::device_filter &Filter : FilterList->get()) {
```

Comment:
Won't it return true (in error) if given index was specified in **any** unrelated triple in the SYCL_DEVICE_FILTER?

Reply:
I thought about the necessity for giving only the designated device_type, but I chose not to.
Yes, the bonus point (30pt) will be given to all devices that are listed in SYCL_DEVICE_FILTER.
Note that 30pt is smaller than any other heuristic point difference given to device_type differences.
But, this small point is useful to break a tie if multiple devices of the same type are available, but it shouldn't affect the decision to select the right device type. Here's why.

1. If the user specified for one device, but there are multiple devices available with the same device type, then the bonus point on the specific device number helps. Giving 30 points to other devices listed in SYCL_DEVICE_TYPE won't change the total relative difference in heuristic points. For example, the heuristic difference between A and B are 100, but giving 30 to each still remain 100 point difference.

2. If only one device type has specified the device number but other device types don't (e.g., A, B: b:2), then the heuristic point difference between A and B is already bigger than 50 points. Giving 30 points won't affect the decision.

3. If the user specified two different device numbers for the same type (For example, SYCL_DEVICE_FILTER=A:  a:0,A: a,1) we can't determine which one should get the best match point if those devices are the same backend and same device_type.

4. If two different devices numbers are specified for different device_type or backend (for example, A: a:0, B: b:1) then based on current heuristic, whichever has the bigger heuristic point will be chosen. 30 points won't affect the decision.

Reply:
I will just change the implementation to give 30 points to only designated device type.

Reply:
Done.